src/strands/tools/registry.py

```diff
@@ -176,7 +169,12 @@ def get_all_tools_config(self) -> Dict[str, Any]:
         logger.debug("tool_count=<%s> | tools configured", len(tool_config))
         return tool_config
 
-    def register_tool(self, tool: AgentTool) -> None:
+    # mypy has problems converting between DecoratedFunctionTool <-> AgentTool
+    @overload
+    def register_tool(self, tool: DecoratedFunctionTool) -> None: ...
+    @overload
+    def register_tool(self, tool: AgentTool) -> None: ...
+    def register_tool(self, tool: AgentTool) -> None:  # type: ignore
```

Comment:
This is interesting, why is `def register_tool` needed 3 times here ^^ 🤔

Reply:
I was going to say: 

> For some reason, mypy doesn't correctly deduce that `DecoratedFunctionTool` derives from `AgentTool` even though it's defined as such - I believe because of the generics

But that made me take another look at it - I was importing from `strands.types.tools` instead of `..types.tools` which was leading to the type error. So removed.  

Thanks!